spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java

```diff
@@ -133,32 +137,15 @@ else if (this.flywayDataSource != null) {
 			else {
 				flyway.setDataSource(this.dataSource);
 			}
+			if (!CollectionUtils.isEmpty(this.flywayCallbacks)) {
+				for (FlywayCallback flywayCallback : this.flywayCallbacks) {
+					flyway.setCallbacks(flywayCallback);
+				}
+			}
 			flyway.setLocations(this.properties.getLocations().toArray(new String[0]));
 			return flyway;
 		}
 
-		@Bean
-		@ConditionalOnMissingBean
```

Comment:
We actually need this. We use it to not execute the migrations on startup by providing a noop `FlywayMigrationStrategy` as described in #4295.